python/pyarrow/tests/test_gpu.py

```diff
@@ -0,0 +1,578 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+UNTESTED:
+CudaDeviceManager.create_new_context
+cuda_read_message
+"""
+
+
+import pytest
+import pyarrow as pa
+import numpy as np
+import sys
+
+gpu_support = pytest.mark.skipif(not pa.has_gpu_support,
+                                 reason='Arrow not built with GPU support')
+
+gpu_ipc_support = pytest.mark.skipif(
+    not sys.platform.startswith(('linux', 'darwin')),
```

Comment:
Does CUDA IPC work on macOS?

Reply:
According to https://github.com/NVIDIA/cuda-samples#cuda-interprocess-communication IPC is available in Linux x86_64 and ppc64le systems.
A more robust check would need to use `sysconfig.get_platform()`.

Reply:
Restricting IPC support to `linux-x86_64` only. PPC64 could be enabled but only after Arrow C++ supports IPC in PPC64.